github-actions-scripts/folioci_npmver.sh

```diff
@@ -1,39 +1,100 @@
-#!/bin/bash
+#!/usr/bin/env node
 
-# for testing only
-#BUILD_NUMBER=123
+/**
+ * SUMMARY
+ *
+ * derive a semver-compatible value from ./package.json's version and the build-number
+ *
+ *
+ * USAGE
+ *
+ * JOB_ID=123 ./$0              // output like 11.1.109000000123
+ * JOB_ID=1234 ./$0             // output like 11.1.109000001234
+ * JOB_ID=123 new_ci=true ./$0  // output like 11.1.10990000000123
+ * JOB_ID=1234 new_ci=true ./$0 // output like 11.1.10990000001234
+ *
+ *
+ * DETAILS
+ *
+ * Given package.json contains a version value like `1.2.0` or `2.3.4` and
+ * the environment variable JOB_ID contains a build-number like 456, concoct
+ * a version number like `1.2.109000000456` or `2.3.409000000456` and print it.
```

Comment:
Should `1.2.109000000456` begin `1.2.0` to reflect the version number `1.2.0` two lines higher in the comment?

Reply:
Nope. When the patch version is `0`, the algorithm blithely changes it to `1` because `009000000456` is not a valid number according semver. JS, of course, is happy to interpret a number with a leading 0 as base 8, but that's not what we want here. 

IOW, this comment deliberately documents an edge case but without saying so, and therefore it looks like the comment is wrong. That's a good point. I'll add a clarification here.